azure-pipelines-20230430.yml

```diff
@@ -255,8 +255,23 @@ stages:
               repository: 'apachehudi/hudi-ci-bundle-validation-base'
               command: 'run'
               arguments: >
+                -v $(Build.SourcesDirectory):/hudi
                 -i docker.io/apachehudi/hudi-ci-bundle-validation-base:$(Build.BuildId)
-                /bin/bash -c "mvn clean install $(MVN_OPTS_INSTALL) -Phudi-platform-service -Pthrift-gen-source
+                /bin/bash -c "pwd
+                && rm -rf /hudi/scripts/ci/results
+                && mvn clean install $(MVN_OPTS_INSTALL) -Phudi-platform-service -Pthrift-gen-source
                 && mvn test  $(MVN_OPTS_TEST) -Punit-tests -pl $(JOB5_UT_MODULES)
-                && mvn test  $(MVN_OPTS_TEST) -Pfunctional-tests -pl $(JOB5_UT_MODULES)
-                && grep \"testcase\" */target/surefire-reports/*.xml */*/target/surefire-reports/*.xml | awk -F'\"' ' { print $6,$4,$2 } ' | sort -nr | head -n 100"
+                && mvn test  $(MVN_OPTS_TEST) -Pfunctional-tests -pl $(JOB5_FT_MODULES)
+                && ./scripts/ci/move_surefire_reports.sh /hudi /hudi/scripts/ci/results
```

Comment:
Curious: since we use `-v` to map the source to /hudi, do we still need to copy the test result?

Reply:
Unfortunately yes, because the tests also add `hudi-utilities/null/hive` folder and extra files for HMS and other stuff, causing the publish job to fail due to file access permissions (these are created by test Hive service which has different read permissions).  So I have to move the surefire report files to a separate directory and let the publish job to look for that directory only to get rid of the permission issue.

Reply:
can you add these details in comments inline.